### PR TITLE
Mayhem

### PR DESCRIPTION
**Pull request by zwimer**

### CODE REVIEW

File: base.docker [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM fuzzers/cargo-fuzz:0.10.0
```
Comment (xansec) [inline base.docker:1]: Having two Dockerfiles is a little messy - can you use a multistage build instead, reducing image size with a package stage?

File: base.docker [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+RUN rustup update
+
+# Source
+RUN git clone https://github.com/tectonic-typesetting/tectonic
+RUN cd tectonic && git submodule update --init
+
+# Compile
+WORKDIR /tectonic/fuzz
+RUN sed -i '$ d' ./run-fuzzer.sh
+RUN echo "cargo fuzz build" >> ./run-fuzzer.sh
```
Comment (xansec) [inline base.docker:26]: I looked at `run-fuzzer.sh ` and I don't think you need to use this script - you should be able to just install and run `cargo fuzz build <target>` directly.

File: Mayhemfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+image: ghcr.io/zwimer/tectonic:mayhem
```
Comment (xansec) [inline Mayhemfile:1]: You can omit the image name.

File: Mayhemfile [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+duration: 90
+project: mayhem-targets
+target: fuzzme
```
Comment (xansec) [inline Mayhemfile:4]: Please ensure your target name and binary/cmd name match.

**Review by xansec — CHANGES_REQUESTED** [review]

File: base.docker (hunk 1 above)
Comment (zwimer) [reply base.docker:1]: Sure

File: Mayhemfile (hunk 3 above)
Comment (zwimer) [reply Mayhemfile:1]: Ok

File: Mayhemfile [hunk 5]
```diff
@@ -0,0 +1,8 @@
+image: ghcr.io/zwimer/tectonic:mayhem
+duration: 90
+project: mayhem-targets
```
[the remark this answers is not in the document]
Comment (zwimer) [reply Mayhemfile:3]: Changed to `tectonic`

File: Mayhemfile (hunk 4 above)
Comment (zwimer) [reply Mayhemfile:4]: Ok

File: base.docker (hunk 1 above)
Comment (zwimer) [reply base.docker:1]: Done

File: base.docker [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	zlib1g-dev \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# Rust
+RUN rustup default nightly
+RUN rustup update
+
+# Source
+RUN git clone https://github.com/tectonic-typesetting/tectonic
```
[the remark this answers is not in the document]
Comment (zwimer) [reply base.docker:20]: Ok

File: base.docker (hunk 2 above)
Comment (zwimer) [reply base.docker:26]: We can copy the contents to the dockerfile instead of reusing the file.

**Comment by r00tus3r** [comment]
@xansec ping, do these changes look good?

**Comment by r00tus3r** [comment]
@xansec Done!

**Comment by xansec** [comment]
I added a few more updates, but this is good. Merging.

**Pull request merged**